src/smtml/interpret.ml

```diff
@@ -26,10 +26,17 @@ module Make (Solver : Solver_intf.S) = struct
       state
     | Check_sat assumptions ->
       Log.debug (fun k -> k "check-sat: %a" Expr.pp_list assumptions);
-      ( match Solver.check solver assumptions with
+      let actual = Solver.check solver assumptions in
+      ( match actual with
       | `Sat -> Fmt.pr "sat@."
       | `Unsat -> Fmt.pr "unsat@."
       | `Unknown -> Fmt.pr "unknown@." );
+      ( match (state.status, actual) with
+      | Some `Sat, `Unsat ->
+        Fmt.failwith "Expected status: sat, but solver returned unsat"
+      | Some `Unsat, `Sat ->
+        Fmt.failwith "Expected status: unsat, but solver returned sat"
```

Comment:
Maybe something like this for the flag suggestion. You'd pass it through main until this function

```suggestion
        if no_strict_status then Log.error ...
        else  Fmt.failwith "Expected status: unsat, but solver returned sat"
```